src/map/lua/lua_baseentity.cpp

```diff
@@ -3632,6 +3632,26 @@ bool CLuaBaseEntity::addLinkpearl(std::string const& lsname, bool equip)
     return false;
 }
 
+auto CLuaBaseEntity::addSoulPlate(std::string const& name, uint16 skillIndex, uint8 fp) -> std::optional<CLuaItem>
+{
+    XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
+    {
+        CItem* PItem = itemutils::GetItem(2477); // Used Soul Plate
```

Comment:
I was thinking we could handle the removal of the blank plate from the ranged slot here as well. the ammo removal code can be a guide.

Reply:
actually can be as short as
```cpp
// Deduct blank plate
charutils::UpdateItem(PChar, loc, slot, -1);
PChar->pushPacket(new CInventoryFinishPacket());
```
+2 lines to get your ammo slot from